src/main/java/domain/LadderResult.java

```diff
@@ -0,0 +1,43 @@
+package domain;
+
+import domain.ladder.Ladder;
+import domain.ladder.Line;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LadderResult {
+
+    private final Ladder ladder;
+    private final List<String> gameResults;
+
+    public LadderResult(Ladder ladder, List<String> gameResults) {
+        this.ladder = ladder;
+        this.gameResults = gameResults;
+    }
+
+    public List<String> calculateResults() {
+
+        int width = ladder.getWidth();
+        List<String> results = new ArrayList<>();
+
+        for (int i = 0; i < width; i++) {
+            int endPoint = calculateEndPoint(i);
+            results.add(gameResults.get(endPoint));
+        }
+        return results;
+    }
+
+    private int calculateEndPoint(int startPoint) {
+        int currentPoint = startPoint;
+
+        for (Line line : ladder.getLines()) {
+            if ((currentPoint < line.getPoints().size() - 1) && (line.getPoints().get(currentPoint))) {
```

Comment:
현재 위치에서 line에 대한 움직임 여부를 판단하고 있는데요.
코드가 요구사항의 어떤 조건을 설명하는지 판단하기 어려워 보여요.
메서드를 분리해서 위 코드가 어떤 조건인지 설명하면 좋을 것 같습니다.

이 역할을 Line에 넘겨주어도 좋아보여요.
```java
// Line.class

public boolean isPointsInRange(int point) {
// ...
};

public boolean isAbleToMoveLeft(int point) {
// ...
}
```


```suggestion
            if (line.isPointsInRange(currentPoint) && line.isAbleToMoveLeft(currentPoint)) {
```
이렇게 보면 어떤 조건에서 currentPoint가 증가 하는지 쉽게 확인할 수 있겠죠?

객체의 값을 가져와서 수행하기 보단, 해당 책임을 가진 객체가 수행할 수 있게 해주세요